src/main/kotlin/net/atos/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -0,0 +1,400 @@
+        )
+        if (zaaktypeUUID.isPresent) {
+            try {
+                zaaktypeUUID.get().let {
+                    LOG.fine { "Creating a zaak using a CMMN case with zaaktype: $it" }
+                    registreerZaakMetCMMNCase(it, productaanvraag, productaanvraagObject)
+                }
+            } catch (exception: RuntimeException) {
+                warning("CMMN", productaanvraag, exception)
+            }
+        } else {
+            findZaaktypeByIdentificatie(productaanvraag.type)?.let {
+                try {
+                    LOG.fine { "Creating a zaak using a BPMN proces with zaaktype: $it" }
+                    registreerZaakMetBPMNProces(it, productaanvraag, productaanvraagObject)
+                } catch (exception: RuntimeException) {
+                    warning("BPMN", productaanvraag, exception)
+                }
+            } ?: run {
+                LOG.info(
+                    message(
+                        productaanvraag,
+                        "No zaaktype found for productaanvraag-Dimpact type '${productaanvraag.type}'. No zaak was created."
+                    )
+                )
+                registreerInbox(productaanvraag, productaanvraagObject)
+            }
+        }
+    }
+
+    private fun warning(processType: String, productaanvraag: ProductaanvraagDimpact, exception: RuntimeException) {
+        LOG.log(
+            Level.WARNING,
+            message(productaanvraag, "Failed to create a zaak of process type: $processType."),
+            exception
+        )
+    }
+
+    private fun message(productaanvraag: ProductaanvraagDimpact, message: String) =
+        "Productaanvraag ${productaanvraag.aanvraaggegevens}: $message"
+
+    private fun registreerZaakMetBPMNProces(
+        zaaktype: ZaakType,
+        productaanvraag: ProductaanvraagDimpact,
+        productaanvraagObject: ORObject
+    ) {
+        val createdZaak = Zaak().apply {
+            this.zaaktype = zaaktype.url
+            bronorganisatie = ConfiguratieService.BRON_ORGANISATIE
+            verantwoordelijkeOrganisatie = ConfiguratieService.BRON_ORGANISATIE
+            startdatum = LocalDate.now()
+        }.let(zgwApiService::createZaak)
+        bpmnService.readProcessDefinitionByprocessDefinitionKey(BPMN_PROCESS_DEFINITION_KEY).let {
+            zrcClientService.createRol(creeerRolGroep(it.description, createdZaak))
+        }
+        pairProductaanvraagInfoWithZaak(productaanvraag, productaanvraagObject, createdZaak)
+        bpmnService.startProcess(
+            createdZaak,
+            zaaktype,
+            getFormulierData(productaanvraagObject),
+            BPMN_PROCESS_DEFINITION_KEY
+        )
+    }
+
+    fun getFormulierData(productaanvraagObject: ORObject): Map<String, Any> {
+        val formulierData = mutableMapOf<String, Any>()
+        (productaanvraagObject.record.data[PRODUCT_AANVRAAG_FORMULIER_DATA_VELD] as Map<*, *>)
+            .forEach { (_, velden) ->
```

Comment:
Do you think this would read better?
```suggestion
            .values.forEach {
```

Reply:
done (will look at this function later anyway.. it is quite messy now)